sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/job/SecHubJobRepositoryImpl.java

```diff
@@ -14,29 +15,90 @@
 import com.daimler.sechub.sharedkernel.jpa.TypedQuerySupport;
 
 public class SecHubJobRepositoryImpl implements SecHubJobRepositoryCustom {
-	/* @formatter:off */
-	public static final String JPQL_STRING_SELECT_BY_EXECUTION_STATE = 
-			"select j from "+CLASS_NAME+" j"+
-					" where j."+PROPERTY_EXECUTION_STATE+" = :"+PROPERTY_EXECUTION_STATE +
-					" order by j."+PROPERTY_CREATED;
-	/* @formatter:on */
-
-	private final TypedQuerySupport<ScheduleSecHubJob> typedQuerySupport = new TypedQuerySupport<>(ScheduleSecHubJob.class);
-
-	@PersistenceContext
-	private EntityManager em;
-
-	@Override
-	public Optional<ScheduleSecHubJob> findNextJobToExecute() {
-
-		Query query = em.createQuery(JPQL_STRING_SELECT_BY_EXECUTION_STATE);
-		query.setParameter(PROPERTY_EXECUTION_STATE, READY_TO_START);
-		query.setMaxResults(1);
-		// we use OPTIMISTIC_FORCE_INCREMENT write lock - so only one POD will be able to execute next job...
-		// see https://www.baeldung.com/jpa-pessimistic-locking
-		query.setLockMode(LockModeType.OPTIMISTIC_FORCE_INCREMENT);
-
-		return typedQuerySupport.getSingleResultAsOptional(query);
-	}
+    /* @formatter:off */
+	static final String JPQL_STRING_SELECT_BY_EXECUTION_STATE = 
+			"select j from " + CLASS_NAME + " j" +
+			" where j." + PROPERTY_EXECUTION_STATE + " = :" + PROPERTY_EXECUTION_STATE +
+			" order by j." + PROPERTY_CREATED;
+	
+	static final String JPQL_STRING_SELECT_BY_JOB_ID = 
+            "select j from " + CLASS_NAME + " j" +
+            " where j." + PROPERTY_EXECUTION_STATE + " = :" + PROPERTY_EXECUTION_STATE +
+            " and j." + PROPERTY_UUID + " = :" + PROPERTY_UUID;
+    
+	
+	static final String SUB_JPQL_STRING_SELECT_PROJECTS_WITH_RUNNING_JOBS = 
+	        "select p. " + PROPERTY_PROJECT_ID + " from " + CLASS_NAME + " p" +
```

Comment:
Only `project_id`s are selected from the underlying `ScheduleSecHubJob`s. I chose `p`, because this is a sub-query of `JPQL_STRING_SELECT_JOB_WHERE_NOT_YET_RUNNING_SAME_PROJECT` where only `projectId`s are selected.